packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.controller.js

```diff
@@ -174,9 +182,24 @@ export default /* @ngInject */ function IpGameFirewallCtrl(
     );
   }
 
-  function init() {
-    getFirewall();
-    getRules();
+  function init(params) {
+    if (params) {
+      self.datas.selectedBlock = params.ipBlock.ipBlock;
+      self.datas.selectedIp = params.ip.ip;
+      getFirewall();
+      getRules();
+    } else {
+      self.datas.selectedIp = $stateParams.ip;
+      $http
+        .get(`/ip?ip=${window.encodeURIComponent($stateParams.ip)}`)
```

Comment:
**niptick** : set `id` params in the `resolve` part of the routing instead of injecting it directly in the controller